kornia/augmentation/tormentor_minimal/spatial_augmentations.py

```diff
@@ -0,0 +1,134 @@
+@aug_statenames("rotate_radians")
+@aug_parameters(min_angle=-math.pi, max_angle=math.pi)
+class Rotate(SpatialImageAugmentation):
+    def forward_batch(self, tensor_image):
+        self.rotate_radians = (torch.rand(tensor_image.size(0)) + self.min_angle)*(self.max_angle-self.min_angle)
+        rotate_degrees = (self.rotate_radians * 180) / math.pi
+        print(rotate_degrees)
+        return kornia.geometry.rotate(tensor_image, rotate_degrees)
+
+    def forward_sample(self, tensor_image):
+        tensor_image = tensor_image.unsqueeze(dim=0)
+        self.rotate_radians = (torch.rand(1) + self.min_angle)*(self.max_angle-self.min_angle)
+        rotate_degrees = (self.rotate_radians * 180) / math.pi
+        print(rotate_degrees)
+        return kornia.geometry.rotate(tensor_image, rotate_degrees)[0, :, :, :]
+
+
+@aug_parameters(min_scale=.5, max_scale=2.0, joint=False)
+class Scale(SpatialImageAugmentation):
+    """Implementation of augmentation by scaling images.
+    """
+    def forward_sample(self, tensor_image):
+        tensor_image.unsqueeze(dim=0)
+        if self.joint:
+            scale = (torch.rand(1) * (self.max_scale - self.min_scale) + self.min_scale).item()
+            result = F.interpolate(tensor_image, scale_factor=[scale, scale], mode='bilinear', align_corners=True)
+        else:
+            width_height_ratios = (torch.rand(2) * (self.max_scale - self.min_scale) + self.min_scale)
+            result = F.interpolate(tensor_image, scale_factor=width_height_ratios, mode='bilinear', align_corners=True)
+        return result[0, :, :, :]
+
+@aug_parameters(horizontal_prob=.5, vertical_prob=.5)
+class Flip(SpatialImageAugmentation):
+    def forward_sample(self, tensor_image):
+        dim_thresholds = torch.Tensor([1.0, 1.0, self.horizontal_prob, self.vertical_prob])
+        rnd = torch.rand(4)
+        flip_dims = torch.nonzero(rnd > dim_thresholds).view(-1)
+        flip_dims = tuple(flip_dims.tolist())  # TODO (anguelos) can we use a tensor instead of a list
+        return tensor_image.flip(dims=flip_dims)
+
+@aug_parameters(min_width=.1, max_width=.5, min_height=.1, max_height=.5)
+class EraseRectangle(SpatialImageAugmentation):
+    def forward_sample(self, tensor_image):
+        result = tensor_image.copy()
+        _, _, image_width, image_height = tensor_image.size()
+        size_minima = torch.Tensor([self.min_width, self.min_height])
+        size_ranges = torch.Tensor([self.max_width - self.min_width, self.max_height - self.min_height])
+        patch_width, patch_height = (torch.rand(2) * size_ranges + size_minima).tolist()
+        left, top = (torch.rand(2)).tolist()
+        left = math.round(image_width - patch_width) * left
+        top = math.round(image_height - patch_height) * top
+        result[:, :, left:left+patch_width, top:top+patch_height] = 0
+        return result
+
+@aug_parameters(desired_width=224, desired_height=224)
+class CropPadAsNeeded(SpatialImageAugmentation):
```

Comment:
To handle the padding, I was using:
```python
        if self.pad_if_needed and input.shape[-2] < self.size[0]:
            padding = [0, 0, (self.size[0] - input.shape[-2]), self.size[0] - input.shape[-2]]
            input = F.pad(input, padding, value=self.fill, mode=self.padding_mode)

        if self.pad_if_needed and input.shape[-1] < self.size[1]:
            padding = [self.size[1] - input.shape[-1], self.size[1] - input.shape[-1], 0, 0]
            input = F.pad(input, padding, value=self.fill, mode=self.padding_mode)
```
I think it looks more cleaner.

Reply:
Actually pad_if_needed was my only contribution to torchvision :D.
Whether pad_if_needed is a switch or a different augmentation to me seems secondary. If torchvision compatibillity is desired, I guess pad_if_needed is better.
Should everything be padded the same way? If augmenting, ideally padding borders should be random adding up to what is needed.